### PR TITLE
signed header generation moved to SignatureService

### PR DESCRIPTION
**Pull request by daita**
Moving the generation of the header to sign a request from ActivityService to SignatureService

### CODE REVIEW

**Comment by daita** [comment]
Also, fixing #238 by filling the header, and checking algorithm on incoming request

**Comment by jancborchardt** [comment]
cc @annando for review as this fixes the issue https://github.com/nextcloud/social/issues/238 you reported :)

File: lib/Service/SignatureService.php [hunk 1]
```diff
@@ -225,9 +269,15 @@ private function checkRequestSignature(IRequest $request): string {
 		$signed = base64_decode($sign['signature']);
 		$estimated = $this->generateEstimatedSignature($headers, $request);
 
+		$algorithm = 'sha256';
+		if ($this->get('algorithm', $sign, '') === 'rsa-sha256') {
```
Comment (annando) [inline lib/Service/SignatureService.php]: Possibly you could check for ```'rsa-sha512'``` here as well, like we do it here:
https://github.com/friendica/friendica/blob/2018.12-rc/src/Util/HTTPSignature.php#L421-L427

Comment (daita) [reply lib/Service/SignatureService.php]: should work now

**Pull request merged**